src/build/arg/mod.rs

```diff
@@ -4808,6 +4808,16 @@ impl<'help> Arg<'help> {
                 self.num_vals = Some(val_names_len);
             }
         }
+
+        let self_id = self.id.clone();
+        if self.is_positional() || self.is_set(ArgSettings::MultipleOccurrences) {
+            // Remove self-overrides where they don't make sense.
+            //
+            // We can evaluate switching this to a debug assert at a later time (though it will
+            // require changing propagation of `AllArgsOverrideSelf`).  Being conservative for now
+            // due to where we are at in the release.
```

Comment:
Now that we released, can we add a debug assert for this?

Reply:
I think doing so would be a breaking change.  This was written assuming we wouldn't be changing it for 3.x because it didn't seem worth addressing in our current time frame.